### ISSUE TITLE
is checkstyle.xml being used during the local build & CI?

### ISSUE THREAD

**Issue by sming** [spotify/heroic#701]
[this PR](https://github.com/spotify/heroic/pull/700) appears to have code that [should perhaps fail checkstyle](https://github.com/spotify/heroic/pull/700/files/772e5948c8c912147d778cd5080121776455a148#diff-17ba0daa8a59cc446bc0bd9ad3419fcdR41), e.g. [here](https://github.com/spotify/heroic/pull/700#discussion_r501843060) but the build permitted it. 

We should double-check that `./checkstyle.xml` in the heroic repo is being used correctly during both the local build and CI.

**Comment by sming** [comment]
So @ao2017 queried that a recent commit I made to cHeckstyle.xml could have broken checkstyle. [The change in question](https://github.com/spotify/heroic/commit/0fe2ca8b511a690db2a5cf9a4bb13c3c71552187) however was just an added explanatory comment so that's not a factor.

**Comment by sming** [comment]
So the next step is to check if CI is broken by committing an egregious checkstyle error to a PR.

**Comment by sming** [comment]
I spent an hour or two battling with localhost checkstyle "long line" issues. Checkstyle simply would not respect any of the suppression directives I could find. Perhaps an IntelliJ bug?

**Comment by sming** [comment]
TL;DR: YES, TLD/checkstyle.xml IS being enforced during CI, just not with the rules we thought it had.
- - -
The error in CI proves that ${basedir}/tools/java.header is being enforced, which in turn proves that ${basedir}/checkstyle.xml is being enforced in CI. Here's the error that I invoked by moving the package declaration above the Spotify legalise header text:

```log
:heroic-core:checkstyleMain (Thread[Daemon worker,5,main]) started.

> Task :heroic-core:checkstyleMain
Excluding []
Task ':heroic-core:checkstyleMain' is not up-to-date because:
  No history is available.
[ant:checkstyle] /home/circleci/project/heroic-core/src/main/java/com/spotify/heroic/http/query/QueryResource.java:1: error: Line does not match expected header line of '/*'.
```

This in fact tallies with what @malish8632 , @ao2017 and I observed as part of @ao2017's PR review. This is because "incorrect" indentation is not flagged by our checkstyle ruleset. Locally I found that the below inconsistently-indented code was not flagged: 

```java

import javax.ws.rs.core.MediaType;
    import org.apache.commons.lang3.tuple.Triple; // wrong indent

// too many lines between statements


@Path("query")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
public class QueryResource {
        private final JavaxRestFramework httpAsync;   // wrong indent
    private final QueryManager query;
```

**Comment by sming** [comment]
Answer: YES, see above comment.
